### PR TITLE
Documentation for issue 1562 (Accessing escaped key with on-demand API)

### PR DESCRIPTION
**Pull request by lemire**
Fixes https://github.com/simdjson/simdjson/issues/1562 with better documentation.

### CODE REVIEW

**Comment by lemire** [comment]
@jkeiser It was documented but people often want examples.

File: doc/basics.md [hunk 1]
```diff
@@ -329,7 +344,7 @@ void recursive_print_json(T&& element) {
         cout << ",";
       }
       // key() returns the unescaped key, if we
```
Comment (marmeladema) [inline doc/basics.md:346]: Maybe rephrase it to `key() returns the original key` or `key() returns the key as specified in the document`?

Otherwise the sentence is weird: `key() returns the unescaped key, if we want the unescaped key, ..`

**Comment by lemire** [comment]
@marmeladema I agree with the importance of good documentation and I appreciate all constructive feedback.

Please keep on providing feedback on what you think can be improved. I always take it into account.

File: doc/basics.md (hunk 1 above)
Comment (lemire) [reply doc/basics.md]: I have improved the wording. It was terrible, thanks.

**Review by jkeiser — APPROVED** [review]
Agreed, examples for this seem pretty necessary! Nice :)

**Comment by lemire** [comment]
Merged.
